lib/src/value/calculation.dart

```diff
@@ -223,6 +607,63 @@ class SassCalculation extends Value {
   /// simplification.
   SassCalculation._(this.name, this.arguments);
 
+  // Returns [value] coerced to [number]'s units.
+  static SassNumber _matchUnits(double value, SassNumber number) {
+    return SassNumber.withUnits(value, numeratorUnits: number.numeratorUnits);
+  }
+
+  /// Returns a rounded [number] based on a selected rounding [strategy],
+  /// to the nearest integer multiple of [step].
+  static SassNumber _roundWithStep(
+      String strategy, SassNumber number, SassNumber step) {
+    if (!{'nearest', 'up', 'down', 'to-zero'}.contains(strategy)) {
+      throw ArgumentError(
+          "$strategy must be either nearest, up, down or to-zero.");
+    }
+
+    if ((number.value.isInfinite && step.value.isInfinite) ||
+        ((step.value == 0 || number.value.isNaN) || step.value.isNaN)) {
```

Comment:
A lot of extra parentheses:

```suggestion
    if ((number.value.isInfinite && step.value.isInfinite) ||
        step.value == 0 || number.value.isNaN || step.value.isNaN) {
```